### PR TITLE
Remove some uses of overload

### LINKED ISSUE #432: Use typevar with values rather than overloading in 'struct' stubs
It seems that all overloads in `stubs/3.2/struct.py` can be refactored to use `AnyStr`, which would be much cleaner.

### PR DESCRIPTION
**Pull request by kirbyfan64**
This fixes #432 and removes some other uses of `@overload` that were pretty simple.

### CODE REVIEW

**Comment by kirbyfan64** [comment]
Yeah, I had a bunch of errors, too. I'm fixing them...

**Comment by kirbyfan64** [comment]
What the heck? I think I managed to crash mypy!? Those Travis build results are screwed up...

File: stubs/3.2/re.py [hunk 1]
```diff
@@ -46,20 +46,12 @@ def findall(pattern: AnyStr, string: AnyStr,
 def finditer(pattern: AnyStr, string: AnyStr,
              flags: int = 0) -> Iterator[Match[AnyStr]]: pass
 
-@overload
-def sub(pattern: AnyStr, repl: AnyStr, string: AnyStr, count: int = 0,
-        flags: int = 0) -> AnyStr: pass
-@overload
-def sub(pattern: AnyStr, repl: Callable[[Match[AnyStr]], AnyStr],
+def sub(pattern: AnyStr, repl: Union[AnyStr, Callable[[Match[AnyStr]]]],
```
Comment (spkersten) [inline stubs/3.2/re.py]: I think you're missing the return type here for `Callable`. I think this is causing the failing tests.

File: stubs/3.2/re.py [hunk 2]
```diff
@@ -46,20 +46,12 @@ def findall(pattern: AnyStr, string: AnyStr,
...
         string: AnyStr, count: int = 0, flags: int = 0) -> AnyStr: pass
```
Comment (spkersten) [inline stubs/3.2/re.py]: The `string` argument needs a default too.

File: stubs/3.2/re.py [hunk 3]
```diff
@@ -46,20 +46,12 @@ def findall(pattern: AnyStr, string: AnyStr,
...
-@overload
-def subn(pattern: AnyStr, repl: AnyStr, string: AnyStr, count: int = 0,
-         flags: int = 0) -> Tuple[AnyStr, int]: pass
-@overload
-def subn(pattern: AnyStr, repl: Callable[[Match[AnyStr]], AnyStr],
-         string: AnyStr, count: int = 0,
-         flags: int = 0) -> Tuple[AnyStr, int]: pass
+def subn(pattern: AnyStr, repl: Union[AnyStr, Callable[[Match[AnyStr]]]],
+         string: AnyStr, count: int = 0, flags: int = 0) -> Tuple[AnyStr, int]:
+    pass
```
Comment (spkersten) [inline stubs/3.2/re.py]: See my comments on `sub`.

**Comment by spkersten** [comment]
Of course, mypy shouldn't crash in this way as a result of a missing argument. I've made issue #673 for this.

**Comment by kirbyfan64** [comment]
@spkersten Thanks! I didn't realize that!

What exactly do you mean by a default argument for `string`? `re.sub` doesn't have a default there...

**Comment by JukkaL** [comment]
Wow, a single PR causing two mypy crashes! You found some code paths that don't have good test coverage.

Narrowed down the second crash: https://github.com/JukkaL/mypy/issues/674

Life is busy currently, but I'll try to find some time this week to debug the issue.

**Comment by JukkaL** [comment]
Fixed #674. Started another build -- hopefully it passes now.

**Comment by spkersten** [comment]
@kirbyfan64 You van ignore the comment about default for string. I was misreading the diff and thought that one of the original signatures didn't have the string argument.

**Comment by JukkaL** [comment]
Looks good! Thanks for the PR!

**Pull request merged**